File: backend/app/schemas/ticket.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class TicketCreate(BaseModel):
    title: str
    title_ar: str | None = None
    description: str | None = None
    category: str
    priority: str = "medium"
    driver_id: uuid.UUID | None = None
    vehicle_id: uuid.UUID | None = None
    assigned_to: uuid.UUID | None = None
    data: dict = {}


class TicketUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    priority: str | None = None
    status: str | None = None
    assigned_to: uuid.UUID | None = None
    data: dict | None = None


class TicketResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    title: str
    title_ar: str | None
    description: str | None
    category: str
    priority: str
    status: str
    driver_id: uuid.UUID | None
    vehicle_id: uuid.UUID | None
    assigned_to: uuid.UUID | None
    created_by: uuid.UUID | None
    data: dict
    resolved_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TicketCommentCreate(BaseModel):
    content: str
    attachments: list = []


class TicketCommentResponse(BaseModel):
    id: uuid.UUID
    ticket_id: uuid.UUID
    user_id: uuid.UUID | None
    content: str
    attachments: list
    created_at: datetime

    model_config = {"from_attributes": True}


class TicketStatsResponse(BaseModel):
    by_status: dict[str, int]
    by_category: dict[str, int]
    by_priority: dict[str, int]
    avg_resolution_hours: float

File: frontend/src/types/ticket.ts
export interface Ticket {
  id: string;
  tenant_id: string;
  title: string;
  title_ar: string | null;
  description: string | null;
  category: string;
  priority: string;
  status: string;
  driver_id: string | null;
  vehicle_id: string | null;
  assigned_to: string | null;
  created_by: string | null;
  data: Record<string, unknown>;
  resolved_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface TicketCreate {
  title: string;
  title_ar?: string;
  description?: string;
  category: string;
  priority?: string;
  driver_id?: string;
  vehicle_id?: string;
  assigned_to?: string;
}

export interface TicketUpdate {
  title?: string;
  description?: string;
  priority?: string;
  status?: string;
  assigned_to?: string;
}

export interface TicketFilters {
  status?: string;
  category?: string;
  priority?: string;
  driver_id?: string;
  assigned_to?: string;
  search?: string;
  page?: number;
  per_page?: number;
}

export interface TicketComment {
  id: string;
  ticket_id: string;
  user_id: string | null;
  content: string;
  attachments: string[];
  created_at: string;
}

export interface TicketCommentCreate {
  content: string;
  attachments?: string[];
}

export interface TicketStats {
  by_status: Record<string, number>;
  by_category: Record<string, number>;
  by_priority: Record<string, number>;
  avg_resolution_hours: number;
}
